lib/ansible/modules/system/java_certs.py

```diff
@@ -0,0 +1,250 @@
+
+def import_cert_path(module, path, keystore_path, keystore_pass, alias):
+    ''' Import certificate from path into keystore located on keystore_path as alias '''
+    import_cmd = ("keytool -importcert -noprompt -keystore '{}' -storepass '{}' "
+                  "-file '{}' -alias '{}'").format(keystore_path, keystore_pass, path, alias)
+
+    # Use local certificate from local path and import it to a java keystore
+    (import_rc, import_out, import_err) = module.run_command(import_cmd, check_rc=False)
+    if import_rc == 0:
+        return module.exit_json(changed=True, msg=import_out,
+                                rc=import_rc, cmd=import_cmd, stdout_lines=import_out,
+                                error=import_err)
+    else:
+        return module.fail_json(msg=import_out, rc=import_rc, cmd=import_cmd)
+
+def delete_cert(module, keystore_path, keystore_pass, alias):
+    ''' Delete cerificate identified with alias from keystore on keystore_path '''
+    del_cmd = ("keytool -delete -keystore '{}' -storepass '{}' "
+               "-alias '{}'").format(keystore_path, keystore_pass, alias)
+
+    # Delete SSL certificate from keystore
+    (del_rc, del_out, del_err) = module.run_command(del_cmd, check_rc=True)
+
+    return module.exit_json(changed=True, msg=del_out,
+                            rc=del_rc, cmd=del_cmd, stdout_lines=del_out,
+                            error=del_err)
+
+def test_keytool(module):
+    ''' Test if keytool is actuall executable or not '''
+    test_cmd = "keytool"
+
+    module.run_command(test_cmd, check_rc=True)
+
+def test_keystore(module, keystore_path):
+    ''' Check if we can access keystore as file or not '''
+    if keystore_path is None:
+        keystore_path = ''
+
+    if not os.path.exists(keystore_path) and not os.path.isfile(keystore_path):
+        ## Keystore doesn't exist we want to create it
+        return module.fail_json(changed=False,
+                                msg="Module require existing keystore at keystore_path '{}'"
+                                .format(keystore_path))
+
+def main():
+    argument_spec = dict(
+        cert_url=dict(required=False, type='str'),
+        cert_path=dict(required=False, type='str'),
+        cert_alias=dict(required=False, type='str'),
+        cert_port=dict(required=False, default='443'),
+        keystore_path=dict(required=False, type='str'),
+        keystore_pass=dict(required=False, default='changeit', type='str'),
+        keystore_create=dict(required=False, default=False, type='bool'),
+        state=dict(required=False, default='present', choices=['present', 'absent'])
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        required_one_of=[['cert_path', 'cert_url']],
+        required_together=[['keystore_path', 'keystore_pass']],
+        mutually_exclusive=[
+            ['cert_url', 'cert_path']
+        ],
+    )
```

Comment:
You can either use `before`/`after`, or you can use `prepared` which is basically the output you want to display. Some modules use the diff format (like +lines and -lines), other modules use other output (like the rsync output format which is suited to display metadata-changes).